woocommerce/payment-gateway/class-sv-wc-payment-gateway-hosted.php

```diff
@@ -446,7 +446,15 @@ protected function process_transaction_response( $order, $response ) {
 			// If approved, payment is complete
 			if ( $response->transaction_approved() ) {
 
-				$order->payment_complete();
+				// determine whether we should complete payment or set to on-hold for later capture
+				if ( $this->supports( self::FEATURE_CREDIT_CARD_AUTHORIZATION ) && $this->perform_credit_card_authorization( $order ) ) {
+
+					$this->mark_order_as_held( $order, $this->supports( self::FEATURE_CREDIT_CARD_AUTHORIZATION ) && $this->perform_credit_card_authorization( $order ) ? esc_html__( 'Authorization only transaction', 'woocommerce-plugin-framework' ) : $response->get_status_message(), $response );
```

Comment:
No need for the if/else here. It's identical to the `if ( $this->supports(...` above so it'll always return true if we got to this point.

Also, `esc_html__()` isn't needed for adding order notes, as that's done on output in the admin.